youtube_dl/extractor/funimation.py

```diff
@@ -0,0 +1,97 @@
+    _USER_AGENT = 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_10_5) AppleWebKit/601.2.7 (KHTML, like Gecko) Version/9.0.1 Safari/601.2.7'
+
+    _TEST = {
+        'url': 'http://www.funimation.com/shows/air/videos/official/breeze',
+        'info_dict': {
+            'id': 'AIRENG0001',
+            'title': 'Air - Breeze',
+            'ext': 'mp4',
+        },
+        'params': {
+            'format': 'lower',
+        }
+    }
+
+    def _real_initialize(self):
+        self._login()
+
+    def _login(self):
+        (username, password) = self._get_login_info()
+        if username is None:
+            return
+        self.report_login()
+        login_url = 'http://www.funimation.com/login'
+        data = urlencode_postdata({
+            'formname': 'loginForm2',
+            'email_field': username,
+            'password_field': password,
+        })
+        login_request = sanitized_Request(login_url, data)
+        login_request.add_header('Content-Type', 'application/x-www-form-urlencoded')
+        try:
+            login = self._download_webpage(login_request, login_url)
+        except ExtractorError as e:
+            if isinstance(e.cause, compat_HTTPError) and e.cause.code == 403:
+                raise ExtractorError('%s is not available in your region.' % self.IE_NAME, expected=True)
+        if re.search(r'<meta property="og:url" content="http://www.funimation.com/login"/>', login) is not None:
+                raise ExtractorError('Unable to login, wrong username or password', expected=True)
+
+    def _real_extract(self, url):
+        mobj = re.match(self._VALID_URL, url)
+        video_id = mobj.group('id')
+        request = sanitized_Request(url)
+        request.add_header('User-Agent', self._USER_AGENT)
+        try:
+            webpage = self._download_webpage(request, url)
+        except ExtractorError as e:
+            if isinstance(e.cause, compat_HTTPError) and e.cause.code == 403:
+                raise ExtractorError('%s is not available in your region.' % self.IE_NAME, expected=True)
+        if re.search(r'"sdUrl":"http', webpage) is None:
+                raise ExtractorError('You are not logged-in or the stream requires subscription.', expected=True)
+        self.report_extraction(url)
+
+        fallback_title = video_id
+        video_base = self._search_regex(r'"sdUrl":"(.+?)(?:-1500K.mp4)"', webpage, 'video_base')
+        video_auth = self._search_regex(r'"authToken":"(.+?)"', webpage, 'video_auth')
+        video_id = self._search_regex(r'"FUNImationID":"(.+?)"', webpage, 'video_id')
+        video_base = video_base.replace('\\', '')
+
+        # lower might be better as the default format, it is the most supported.
+        formats_map = [
+            ('lowest', 480, 1000),
+            ('lower', 480, 1500),
+            ('low', 480, 2000),
+            ('medium', 720, 2500),
+            ('high', 720, 3500),
+            ('highest', 1080, 4000),
```

Comment:
Not all formats are available for all episodes. Trying higher formats says this for me:

    [debug] System config: []
    [debug] User config: []
    [debug] Command-line args: [u'http://www.funimation.com/shows/air/videos/official/breeze', u'--format', u'high', u'--verbose']
    [debug] Encodings: locale US-ASCII, fs utf-8, out US-ASCII, pref US-ASCII
    [debug] youtube-dl version 2015.11.27.1
    [debug] Git HEAD: 4617ebb
    [debug] Python version 2.7.10 - Darwin-15.0.0-x86_64-i386-64bit
    [debug] exe versions: ffmpeg 2.8.3, ffprobe 2.8.3
    [debug] Proxy map: {}
    [Funimation] breeze: Downloading webpage
    [debug] Invoking downloader on u'http://wpc.8c48.edgecastcdn.net/008C48/SV/480/AIRENG0001/AIRENG0001-480-3500K.mp4?Z0QmTAlHnPRKQx5Lj0gl8LsPqtkUGwKoQHCdXsS0mTACzVG8qKJxMyO_Jk3LvQs_XxH1xiqPzKb6J85mlEYB_rKzJgd5-jbPhSOodUFm4IvEfhdHf_Cgb3suGBo3cXzWz7dl20zGy6k2X-6o71vlDXo'
    ERROR: unable to download video data: HTTP Error 404: Not Found

I tried to find a way to report this, but couldn't handle it.